src/scripts/app.js

```diff
@@ -1 +1,58 @@
-console.log('hello');
\ No newline at end of file
+console.log('hello');
+
+// PRODUCT SECTION RATING STAR MECHANISM -- WDP181103-19
+
+const productsSection = document.querySelector(".section--products");
+const starBoxes = document.querySelectorAll(".stars");
+
+productsSection.addEventListener("mouseover", function(e) {
+    if (event.target.classList.contains('star')) {
+        let arr = Object.values(event.relatedTarget.children);
+        let index = arr.indexOf(event.target);
+        desactiveStars();
+        for (i=0; i<=index; i++) {
+            event.relatedTarget.children[i].classList.add("active");
+        }
+    }
+    else if (event.target.classList.contains('stars')) {
+        deselectStars();
+    }   
+});
+
+productsSection.addEventListener("click", function(e) {
+    if (event.target.classList.contains('star')) {
+        let arr = Object.values(event.target.parentElement.children);
+        let index = arr.indexOf(event.target);
+        desactiveStars();
+        unmarkStars();
+        for (i=0; i<=index; i++) {
+            event.target.parentElement.children[i].classList.add("permanent");
+        }
+    }
+});
+
+// deselecting stars
+function deselectStars() {
+    if (event.target.children[0].classList.contains("permanent")){}
```

Comment:
Skoro if nie ma żadnych instrukcji, to zamiast tworzyc else i tam wstawiać wlasciwy kod, mozna by zrobic negację warunku w if.  Wtedy moglabys skrocic ten blok.

Reply:
Dodano negację
`if (! event.target.children[0].classList.contains("permanent"))`